src/core/api/payment_manager_api/restapi-cd/v3/_base_policy.xml.tpl

```diff
@@ -47,10 +47,13 @@
           <!-- Credit card onboarding -->
           <when condition="@( context.Request.Url.Path.EndsWith("/webview/transactions/cc/verify") && (string)context.Request.Headers.GetValueOrDefault("Referer","") == "${payment_wallet_origin}")">
               <set-variable name="walletId" value="@{
-                    string requestBody = ((string)context.Variables["requestBody"]);
-                    string walletParam = requestBody!=null && requestBody.Split('&').Length >= 1 ? requestBody.Split('&')[0] : "";
-                    string walletId = walletParam != null && walletParam.Split('=').Length == 2 ? walletParam.Split('=')[1] : "";
-                    return walletId;
+                string requestBody = ((string)context.Variables["requestBody"]);
+                string[] walletParamArray = requestBody!=null && requestBody.Split('&').Length >= 1 ? requestBody.Split('&') : new string[0];
+                string[] walletIdArray = Array.FindAll(walletParamArray, formIdValue => formIdValue.StartsWith("idWallet=", StringComparison.Ordinal));
```

Comment:
Just one consideration: why not use [Find](https://learn.microsoft.com/en-us/dotnet/api/system.array.find?view=net-8.0) method here instead of FindAll?